### DISCUSSION TITLE
Task bar always ontop

### DISCUSSION THREAD

**Discussion by mrmatthewspiers** [valinet/ExplorerPatcher#2775] [General]
Hey, I am using this software and I found that the taskbar is not always on top of applications, it hides behind them and if it is on top it just goes over part of the window and does not resize the window to work with the taskbar so both can be seen properly, can this be fixed please?

**Comment by pyrates999** [comment]
Please post windows version and build.  You can find this by running winver.
Please post explorer patcher version.

Please include what settings you changed in explorer patcher properties application.  These should be screen shots.

Please include any other software you have installed that modifies the UI of windows.

Please show screen shots of when the taskbar is not on top.

**Reply by mrmatthewspiers** [reply]
hey,

Thanks for getting back to me so quick,

windows 11 version 22H2 OS Build 22621.3007
ExplorerPatcher version 22621.2861.62.2

No other software installed that modifies the UI of windows

I have a 4 screen setup including the laptop screen and have moved the taskbar to the far left of 1 of my screens and have it only showing on that screen, I have not changed any settings, other than to put the bar to the left of the screen.

There are a couple of bugs I have noticed (although I really like this taskbar as I can't stand the windows 11 task bar), 
These are as follows:
1 - When multiple windows are open of the same application, like google for example, it shows multiple icons instead of combining them
2 - when windows are maximised on the screen with the taskbar on, it either goes behind the bar or on top of the bar, but never beside.

![Screenshot 2024-02-03 102330](https://github.com/valinet/ExplorerPatcher/assets/82327218/568b8268-e04a-42ad-9bc4-23ac8c1cf67b)
![Screenshot 2024-02-03 102346](https
![Taskbar is behind google](https://github.com/valinet/ExplorerPatcher/assets/82327218/b81be5fb-0994-4d1b-8854-73d5ce567e74)
://github.com/valinet/ExplorerPatcher/assets/82327218/43878d85-ae37-4427-8b6d-0a76386bb2c0)
![Screenshot 2024-02-03 102359](https://github.com/valinet/ExplorerPatcher/assets/82327218/1a5674b6-5f98-442b-b169-34f218f4b5c5)
![Screenshot 2024-02-03 102412](https://github.com/valinet/ExplorerPatcher/assets/82327218/80cbcab0-cf26-472c-868a-e67c861b56ce)
![Screenshot 2024-02-03 102427](https://github.com/valinet/ExplorerPatcher/assets/82327218/15be2ab9-4764-4196-aa37-c9f219d76673)
![Screenshot 2024-02-03 102436](https://github.com/valinet/ExplorerPatcher/assets/82327218/9e3e83fd-fa1e-4c5f-a00d-bd2bddc87a64)
![Screenshot 2024-02-03 102446](https://github.com/valinet/ExplorerPatcher/assets/82327218/3b1b1bb5-b52b-4436-a2cd-568955890c0a)
![Screenshot 2024-02-03 102505](https://github.com/valinet/ExplorerPatcher/assets/82327218/1856d1bb-0c87-4cce-a98d-1168b244472f)
![Screenshot 2024-02-03 102522](https://github.com/valinet/ExplorerPatcher/assets/82327218/a4d9ceb3-f156-4dd4-b541-6988d9032878)
![Screenshot 2024-02-03 102540](https://github.com/valinet/ExplorerPatcher/assets/82327218/57ad9d92-17eb-40c1-94a4-18e8032e6ae6)
![Screenshot 2024-02-03 102550](https://github.com/valinet/ExplorerPatcher/assets/82327218/845f46e2-5409-45e4-a962-b3ba14512f36)
![Multiple google icons instead of combined](https://github.com/valinet/ExplorerPatcher/assets/82327218/ffa78201-a16e-4f7d-8dde-1147caa963ea)
![Window behind taskbar instead of beside](https://github.com/valinet/ExplorerPatcher/assets/82327218/3d9df043-bf59-4e89-b3cb-712b8d6e0ef4)

**Comment by pyrates999** [comment]
You're missing the taskbar settings in the EP properties application.  Please show that.

**Reply by mrmatthewspiers** [reply]
Sorry, forgot them, here they are
![Screenshot 2024-02-03 113651](https://github.com/valinet/ExplorerPatcher/assets/82327218/76ff4e50-d59b-469a-84ba-7a484ba5bef5)
![Screenshot 2024-02-03 113718](https://github.com/valinet/ExplorerPatcher/assets/82327218/29691589-dbb5-459f-9249-628c94239c9b)
![Screenshot 2024-02-03 113733](https://github.com/valinet/ExplorerPatcher/assets/82327218/d4973ea0-e3ea-4364-8860-39b4cedbb7e6)

**Comment by pyrates999** [comment]
Click here:

![image](https://github.com/valinet/ExplorerPatcher/assets/106135298/06af2185-f431-4c64-ab90-1ce9002fa5d3)

**Reply by mrmatthewspiers** [reply]
sorry lol, misunderstood what you meant
![Screenshot 2024-02-03 124848](https://github.com/valinet/ExplorerPatcher/assets/82327218/8ed71e98-98af-483c-823f-1a828bae1d0a)

**Comment by pyrates999** [comment]
Regarding this: 1 - When multiple windows are open of the same application, like google for example, it shows multiple icons instead of combining them

You have **combine taskbar icons on primary taskbar** and **combine taskbar icons on secondary taskbar(s)** set to **combine when taskbar is full**.  That means icons of the same application may combine sometimes depending on how full the taskbar is.  So not a bug.  If you want it to be combined, change the option to **always combine (default)**.  If you don't want it combined, set it to **never combine**.

As for windows being shown underneath the taskbar when maximized, please for a developer to respond.

**Reply by mrmatthewspiers** [reply]
Brilliant thanks for the help, I did wounder if it was down to how I had it setup, I will give it a go

**Reply by mrmatthewspiers** [reply]
cool, thanks for your help, that worked, changing those setting appears to have fixed the taskbar on top problem too

**Reply by pyrates999** [reply]
Please mark my response as the answer then.

**Comment by mrmatthewspiers** [comment]
Resolved by above fix
